Autopilot/PathManager/Src/waypointManager.cpp

```diff
@@ -5,95 +5,189 @@
  * Last Updated: November 2020 (Dhruv)
  */
 
-#include "waypointManager.hpp"
+#include "../Inc/waypointManager.hpp"
+
+#define LINE_FOLLOWING 0
+#define ORBIT_FOLLOWING 1
+
+
+//Constants
+#define EARTH_RADIUS 6378.137
+#define MAX_PATH_APPROACH_ANGLE M_PI/2
+
+//Basic Mathematical Conversions
+#define deg2rad(angle_in_degrees) ((angle_in_degrees) * M_PI/180.0)
+#define rad2deg(angle_in_radians) ((angle_in_radians) * 180.0/M_PI)
+
+//Waypoint Types
+#define DEFAULT_WAYPOINT 0 // Used to navigate flight path
+#define WAYPOINT_USED 1 // Used by the waypint manager to signal that a waypoint has been used
+#define HOLD_WAYPOINT 2 // Circle
+
+// Latitude and longitude of starting area... will need to change accordingly
```

Comment:
Would you rather these be passed in during initialization of your module ? We don't really want to be changing source code every time we fly in a different area.

Reply:
yeah, you can pass them in. I will just use them to make the code more readable. Because if I write `if (waypointPointer->waypointType == 1){}`, it is a lot less clear than `if (waypointPointer->waypointType == WAYPOINT_USED){}`

Reply:
na you misunderstood my comment. I was reffering to lattitude and longitude of starting area. I don't see how you could know those things at compile time.

(Having macros fir waypoint used and stuff is def better than magic numbers)

Reply:
Oh lol, yes, we should pass it in!